server/events/project_command_builder.go

```diff
@@ -267,15 +271,19 @@ func (p *DefaultProjectCommandBuilder) buildPlanAllCommands(ctx *command.Context
 	}
 
 	var projCtxs []command.ProjectContext
+	var repoCfg valid.RepoCfg
 
 	if hasRepoCfg {
-		// If there's a repo cfg then we'll use it to figure out which projects
+		// If there's a repo cfg with projects then we'll use it to figure out which projects
 		// should be planed.
-		repoCfg, err := p.ParserValidator.ParseRepoCfg(repoDir, p.GlobalCfg, ctx.Pull.BaseRepo.ID())
+		repoCfg, err = p.ParserValidator.ParseRepoCfg(repoDir, p.GlobalCfg, ctx.Pull.BaseRepo.ID())
 		if err != nil {
 			return nil, errors.Wrapf(err, "parsing %s", config.AtlantisYAMLFilename)
 		}
 		ctx.Log.Info("successfully parsed %s file", config.AtlantisYAMLFilename)
+	}
+
+	if hasRepoCfg && len(repoCfg.Projects) > 0 {
```

Comment:
We already checked before for `hasRepoCfg`. Why check again?
```suggestion
		if len(repoCfg.Projects) > 0 {
```

Reply:
Seems like this code needs to be refactored a bit because this is getting harder to read

Reply:
Totally agree with this, though as a first time contributor I wanted to keep the diff simpler.

Reply:
That's ok. I'd rather keep the code simpler than the diff simpler. :)

Reply:
@csainty i didn't mean a heavy refactor, while you could do that, i meant just the suggested change above. The if block above it already checks for hasRepoCfg so the second condition can be a single if inside of the above if instead of creating a brand new if block as you proposed.

Does that make sense?

Reply:
It can't actually be inside the above block due to the else clause. But there is no need to check the variable again, we can just rely on `repoCfg.Projects` being empty if `hasRepoCfg` is `false`